src/Domain/FunctionsOfTime/CMakeLists.txt

```diff
@@ -29,6 +32,7 @@ target_link_libraries(
   ${LIBRARY}
   PUBLIC
   DataStructures
+  IO
```

Comment:
Move this down to `PRIVATE` (it's only used in the cpp)

Also add `Options` (as `INTERFACE` because it's only used in a header).

Looks like you're also using `Domain` and `DomainCreators` (which also depends on `Domain`) in the Tags.. isn't that a circular dependence? I think we have to be a bit more clear here on what dependency graph we're going for with all the domain-related libraries and start writing those decisions down as comments in the cmake files.

Reply:
discussed offline: the circular dependency `FunctionsOfTime < DomainCreators < Domain < FunctionOfTime` was there before this PR, so we should fix it but don't have to do so in this PR